### PR TITLE
feat: Recents enhanced

### PR DESCRIPTION
**Pull request by bzang**
Improvements to recents widget

- [ ] Snapshot tests for redux
- [x] Fetching avatars for spaces and rendering them
- [x] decryption key caching
- [x] Incoming Call scenarios
- [x] Handle active user being added to a room
- [x] Handle events appearing for spaces that haven't been loaded

### CODE REVIEW

File: packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js [hunk 1]
```diff
@@ -27,19 +32,22 @@ export function updateSpacesStatus(status) {
  * @param {Boolean} isSelf if actor is the same as this user
  * @returns {Object} action
  */
-export function updateSpaceWithActivity(activity, isSelf) {
+export function updateSpaceWithActivity(activity, isSelf, isReadable = false) {
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js]: jsdoc needed for `isReadable`

File: packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js [hunk 2]
```diff
@@ -84,40 +77,91 @@ export function removeSpace(id) {
...
+    }
+  };
+}
 
 /**
  * Fetches a list of spaces for the authenticated sparkInstance
  *
  * @export
  * @param {Object} sparkInstance
+ * @param {Function} fetchAvatar
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js]: Is this needed?

Comment (bzang) [inline packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js]: Good catch. Remnants of a test

File: packages/node_modules/@ciscospark/widget-recents/src/events.js [hunk 3]
```diff
@@ -66,12 +78,21 @@ export function constructMembershipEventData(activity) {
     actorId: constructHydraId(`PEOPLE`, activity.actor.id),
     id: constructHydraId(`MESSAGE`, activity.id),
     roomId: constructHydraId(`ROOM`, activity.target.id),
-    personId: constructHydraId(`PEOPLE`, activity.target.id),
-    personEmail: activity.target.emailAddress,
+    personId: constructHydraId(`PEOPLE`, activity.object.id),
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-recents/src/events.js]: I like this whole event construction system in one place. Very nice.

**Comment by adamweeks** [comment]
@bzang heads up, merge conflict on this

**Pull request merged**